apps/src/block_utils.js

```diff
@@ -623,8 +623,10 @@ const STANDARD_INPUT_TYPES = {
   },
   [FIELD_INPUT]: {
     addInput(blockly, block, inputConfig, currentInputRow) {
+      const changeHandler = 'Number' === inputConfig.type ? blockly.FieldTextInput.numberValidator : undefined;
+      const fieldTextInput = new blockly.FieldTextInput('', changeHandler);
```

Comment:
Here's the relevant code snippet in blockly explaining why this works:

https://github.com/code-dot-org/blockly/blob/4300559452ce50bfd29511ca7c90c022eca8cd1e/core/ui/fields/field_textinput.js#L150-L153